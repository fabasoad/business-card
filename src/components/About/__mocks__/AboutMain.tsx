import * as React from 'react'

export default function AboutMain() {
  return (
    <div data-testid="AboutMain"></div>
  )
}
