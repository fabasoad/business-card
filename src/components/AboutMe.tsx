/// <reference path='./AboutMe.d.ts' />
import * as React from 'react';
import { useTranslation } from 'react-i18next';
import * as moment from 'moment';

import imgAboutMe from "../assets/images/about_me.jpg";

export default function AboutMe(props) {
  const { t } = useTranslation();

  const now : moment.Moment = moment();

  const startWorkDate : moment.Moment = moment('2010-03-01', 'YYYY-MM-DD');
  const duration1 : moment.Duration = moment.duration(now.diff(startWorkDate));

  const agileWorkDate : moment.Moment = moment('2011-10-01', 'YYYY-MM-DD');
  const duration2 : moment.Duration = moment.duration(now.diff(agileWorkDate));

  const generalListItem1Keys = {
    "totalExperience": duration1.locale(props.getLocale().code).humanize(),
    "agileExperience": duration2.locale(props.getLocale().code).humanize()
  };

  return (
    <div id="about">
      <div className="container">
        <div className="section-title text-center center">
          <h2>{t('business-card-about-me-title')}</h2>
          <hr />
        </div>
        <div className="row d-flex justify-content-center">
          <div className="col-md-12 text-center">
            <img className="img-thumbnail" src={imgAboutMe} alt="About me" />
          </div>
          <div className="col-md-8 col-md-offset-2 text-center">
            <div className="about-text">
              <p>{t("business-card-about-me-general-title")}</p>
              <ul className="list-unstyled">
                <li>{t("business-card-about-me-general-list-item-1", generalListItem1Keys)}</li>
                <li>{t("business-card-about-me-general-list-item-2")}</li>
                <li>{t("business-card-about-me-general-list-item-3")}</li>
                <li>{t("business-card-about-me-general-list-item-4")}</li>
                <li>{t("business-card-about-me-general-list-item-5")}</li>
                <li>{t("business-card-about-me-general-list-item-6")}</li>
              </ul>
            </div>
          </div>
        </div>
      </div>
    </div>
  );
}