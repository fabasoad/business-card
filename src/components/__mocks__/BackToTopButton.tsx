import * as React from 'react'

export default function BackToTopButton() {
  return (
    <div data-testid="BackToTopButton"></div>
  )
}
