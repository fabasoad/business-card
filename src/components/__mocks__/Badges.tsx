import * as React from 'react'

export default function Badges() {
  return (
    <div data-testid="Badges"></div>
  )
}
