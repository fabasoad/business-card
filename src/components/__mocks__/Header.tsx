import * as React from 'react'

export default function Header() {
  return (
    <div data-testid="Header"></div>
  )
}
