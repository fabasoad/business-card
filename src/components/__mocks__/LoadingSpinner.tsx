import * as React from 'react'
import { LoadingSpinnerProps } from '../LoadingSpinner'

export default function LoadingSpinner({ background, color }: LoadingSpinnerProps) {
  return (
    <div data-testid={`LoadingSpinner-${background}-${color}`}></div>
  )
}
