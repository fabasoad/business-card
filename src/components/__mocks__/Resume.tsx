import * as React from 'react'

export default function Resume() {
  return (
    <div data-testid="Resume"></div>
  )
}
