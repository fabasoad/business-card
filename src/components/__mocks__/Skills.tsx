import * as React from 'react'

export default function Skills() {
  return (
    <div data-testid="Skills"></div>
  )
}
