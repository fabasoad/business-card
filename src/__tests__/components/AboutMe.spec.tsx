/// <reference types="jest" />
import * as React from 'react';
import { shallow } from 'enzyme';
import AboutMe from '../../components/AboutMe';
import SupportedLocales from '../../scripts/SupportedLocales';
import { useTranslation } from '../__mocks__/react-i18next';

test('should render AboutMe correctly', () => {
  const { t } = useTranslation();

  const getLocaleMock = jest.fn(SupportedLocales.default);
  const wrapper = shallow(<AboutMe getLocale={getLocaleMock} />);
  expect(wrapper).toMatchSnapshot();
  expect(getLocaleMock).toBeCalledTimes(2);
  expect(t).toBeCalledTimes(7);

  expect(t).toHaveBeenCalledWith(
    expect.stringMatching(/^business-card-about-me-title$/)
  );
  expect(t).toHaveBeenCalledWith(
    expect.stringMatching(/^business-card-about-me-general-list-item-1$/),
    expect.objectContaining({
      totalExperience: '10 years',
      agileExperience: '9 years'
    })
  );
  for (let i = 2; i <= 6; i++) {
    expect(t).toHaveBeenCalledWith(
      expect.stringMatching(new RegExp(`^business-card-about-me-general-list-item-${i}$`))
    );
  }
});