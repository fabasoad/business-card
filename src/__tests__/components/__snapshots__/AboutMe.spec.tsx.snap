// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render AboutMe correctly 1`] = `
<div
  id="about"
>
  <div
    className="container"
  >
    <div
      className="section-title text-center center"
    >
      <h2>
        business-card-about-me-title
      </h2>
      <hr />
    </div>
    <div
      className="row d-flex justify-content-center"
    >
      <div
        className="col-md-12 text-center"
      >
        <img
          alt="About me"
          className="img-thumbnail"
          src="test-file-stub"
        />
      </div>
      <div
        className="col-md-8 col-md-offset-2 text-center"
      >
        <div
          className="about-text"
        >
          <ul
            className="list-unstyled mt-3"
          >
            <li>
              business-card-about-me-general-list-item-1
            </li>
            <li>
              business-card-about-me-general-list-item-2
            </li>
            <li>
              business-card-about-me-general-list-item-3
            </li>
            <li>
              business-card-about-me-general-list-item-4
            </li>
            <li>
              business-card-about-me-general-list-item-5
            </li>
            <li>
              business-card-about-me-general-list-item-6
            </li>
          </ul>
        </div>
      </div>
    </div>
  </div>
</div>
`;
